connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -2777,7 +2778,11 @@ class Dataset[T] private[sql] (
   }
 
   def toJSON: Dataset[String] = {
-    throw new UnsupportedOperationException("toJSON is not implemented.")
+    sparkSession.newDataset(StringEncoder) { builder =>
```

Comment:
Why is this better over `project(to_json(col(*))).as(StringEncoder)`?

Reply:
Sorry ... what is project?

Reply:
Thanks . Sorry I didn't understand what  `project` is before ...

Reply:
fixed